doc-tool/src/dotty/tools/dottydoc/DocFrontEnd.scala

```diff
@@ -1,17 +1,44 @@
 package dotty.tools
 package dottydoc
 
-import dotc.typer.FrontEnd
+import dotc.fromtasty.ReadTastyTreesFromClasses
+import dotc.typer.{FrontEnd, Typer}
 import dotc.core.Contexts.Context
 import dotc.CompilationUnit
 
+import util.syntax.ContextWithContextDottydoc
+
 /** `DocFrontEnd` uses the Dotty `FrontEnd` without discarding the AnyVal
  *  interfaces for Boolean, Int, Char, Long, Byte etc.
  *
+ *  If `-from-tasty` is set, then the trees and documentation will be loaded
+ *  from TASTY. The comments will be cooked after being unpickled.
+ *
  *  It currently still throws away Java sources by overriding
  *  `discardAfterTyper`.
  */
 class DocFrontEnd extends FrontEnd {
+
+  override def runOn(units: List[CompilationUnit])(implicit ctx: Context): List[CompilationUnit] = {
+    if (ctx.settings.fromTasty.value) {
+      val fromTastyFrontend = new ReadTastyTreesFromClasses
```

Comment:
A front end should probably not be in another frontend.  `DocCompiler` should choose between `DocFrontEnd` or `ReadTastyTreesFromClasses` based on `ctx.settings.fromTasty.value`

Reply:
I also wanted to decouple everything, but I ran into issues when cooking the comments. I'll investigate a bit more.